test/e2e/nftables_test.go

```diff
@@ -69,19 +69,23 @@ var _ = Describe("Nftables", func() {
 				role, utilsHelpers)
 			Expect(err).ToNot(HaveOccurred())
 
-			err = cluster.ApplyMachineConfig(machineConfig, cs)
+			needWaitToMCP, err := cluster.ApplyMachineConfigAndCheckChange(machineConfig, cs)
```

Comment:
Maybe
updated, err := cluster.ApplyMachineConfig(machineConfig, cs)

and add a comment to ApplyMachineConfig that it's returning if updated or not